### ISSUE TITLE
[Bug] RC crashes on start on Linux system

### ISSUE THREAD

**Issue by vlaaad** [UnderwaterApps/overlap2d#81]
I just downloaded o2d-0.1.0-RC.zip and tried to run it with `java -jar overlap2d-0.1.0.jar`. It shows for a little some small window and then crashes.

What's in the console:
```
vlaaad@vlaaad-notebook ~/D/o2d> java -jar overlap2d-0.1.0.jar 
Gtk-Message: Failed to load module "pantheon-filechooser-module"

(java:13383): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(java:13383): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(java:13383): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(java:13383): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(java:13383): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
Gtk-Message: Failed to load module "canberra-gtk-module"
java.lang.IllegalStateException: The Drawable has no context available.
	at org.lwjgl.opengl.DrawableGL.checkDestroyed(DrawableGL.java:152)
	at org.lwjgl.opengl.DrawableGL.makeCurrent(DrawableGL.java:109)
	at org.lwjgl.opengl.Display.makeCurrent(Display.java:706)
	at org.lwjgl.opengl.Display.makeCurrentAndSetSwapInterval(Display.java:1025)
	at org.lwjgl.opengl.Display.setParent(Display.java:452)
	at com.badlogic.gdx.backends.lwjgl.LwjglGraphics.setupDisplay(LwjglGraphics.java:137)
	at com.badlogic.gdx.backends.lwjgl.LwjglCanvas.create(LwjglCanvas.java:188)
	at com.badlogic.gdx.backends.lwjgl.LwjglCanvas$1.addNotify(LwjglCanvas.java:88)
	at java.awt.Container.addImpl(Container.java:1119)
	at java.awt.Container.add(Container.java:415)
	at com.badlogic.gdx.backends.lwjgl.LwjglFrame.addCanvas(LwjglFrame.java:135)
	at com.badlogic.gdx.backends.lwjgl.LwjglFrame$2.run(LwjglFrame.java:96)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:749)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:702)
	at java.awt.EventQueue$3.run(EventQueue.java:696)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:719)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
org.lwjgl.LWJGLException: X Error - disp: 0x7feff83921e0 serial: 246 error: BadWindow (invalid Window parameter) request_code: 2 minor_code: 0
	at org.lwjgl.opengl.LinuxDisplay.globalErrorHandler(LinuxDisplay.java:320)
	at org.lwjgl.opengl.LinuxKeyboard.nSetDetectableKeyRepeat(Native Method)
	at org.lwjgl.opengl.LinuxKeyboard.setDetectableKeyRepeat(LinuxKeyboard.java:152)
	at org.lwjgl.opengl.LinuxKeyboard.destroy(LinuxKeyboard.java:163)
	at org.lwjgl.opengl.LinuxDisplay.destroyKeyboard(LinuxDisplay.java:1264)
	at org.lwjgl.input.Keyboard.destroy(Keyboard.java:357)
	at org.lwjgl.opengl.Display.destroyWindow(Display.java:349)
	at org.lwjgl.opengl.Display.access$400(Display.java:65)
	at org.lwjgl.opengl.Display$5.destroy(Display.java:835)
	at org.lwjgl.opengl.Display.destroy(Display.java:1095)
	at com.badlogic.gdx.backends.lwjgl.LwjglGraphics.createDisplayPixelFormat(LwjglGraphics.java:197)
	at com.badlogic.gdx.backends.lwjgl.LwjglGraphics.setupDisplay(LwjglGraphics.java:174)
	at com.badlogic.gdx.backends.lwjgl.LwjglCanvas.create(LwjglCanvas.java:188)
	at com.badlogic.gdx.backends.lwjgl.LwjglCanvas$1.addNotify(LwjglCanvas.java:88)
	at java.awt.Container.addNotify(Container.java:2773)
	at javax.swing.JComponent.addNotify(JComponent.java:4749)
	at java.awt.Container.addNotify(Container.java:2773)
	at javax.swing.JComponent.addNotify(JComponent.java:4749)
	at java.awt.Container.addNotify(Container.java:2773)
	at javax.swing.JComponent.addNotify(JComponent.java:4749)
	at javax.swing.JRootPane.addNotify(JRootPane.java:757)
	at java.awt.Container.addNotify(Container.java:2773)
	at java.awt.Window.addNotify(Window.java:770)
	at java.awt.Frame.addNotify(Frame.java:487)
	at java.awt.Window.show(Window.java:1031)
	at java.awt.Component.show(Component.java:1654)
	at java.awt.Component.setVisible(Component.java:1606)
	at java.awt.Window.setVisible(Window.java:1014)
	at com.badlogic.gdx.backends.lwjgl.LwjglFrame$2.run(LwjglFrame.java:97)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:749)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:702)
	at java.awt.EventQueue$3.run(EventQueue.java:696)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:719)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at com.uwsoft.editor.Overlap2D.sendNotification(Overlap2D.java:107)
	at com.uwsoft.editor.Overlap2D.pause(Overlap2D.java:75)
	at com.badlogic.gdx.backends.lwjgl.LwjglCanvas$4.run(LwjglCanvas.java:308)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:749)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:702)
	at java.awt.EventQueue$3.run(EventQueue.java:696)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:719)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by azakhary** [comment]
Hmm, thanks for reporting, that's weird, has to be debugged. Can you run from sources?

**Comment by vlaaad** [comment]
I can't compile it :P

```
/home/vlaaad/programming/projects/overlap2d-runtime-libgdx/src/com/brashmonkey/spriter/Calculator.java:22: error: unmappable character for encoding UTF8
	 * @return Smallest difference between a and b (between 180� and -180�).
	                                                           ^
/home/vlaaad/programming/projects/overlap2d-runtime-libgdx/src/com/brashmonkey/spriter/Calculator.java:22: error: unmappable character for encoding UTF8
	 * @return Smallest difference between a and b (between 180� and -180�).
	                                                                     ^
/home/vlaaad/programming/projects/overlap2d-runtime-libgdx/src/com/uwsoft/editor/renderer/components/ZindexComponent.java:5: error: class ZIndexComponent is public, should be declared in a file named ZIndexComponent.java
public class ZIndexComponent extends Component {
       ^

```

**Comment by vlaaad** [comment]
Trying to run it with `./gradlew run`

**Comment by azakhary** [comment]
Hmm I think this is some kind of Linux thing. I just refactored the ZIndexComponent (just changed some letters to upper case) and it seems like for my windows machine it changed the file name too as it should.. Could you check how it looks on yours.

It's linux right?

**Comment by vlaaad** [comment]
Yeah, I'm on linux, thus I have casesensitive file system.
`git pull` says everything is up to date...

**Comment by azakhary** [comment]
Always trouble with Linux :-P I'll check what it is now, give me a minute.

**Comment by vlaaad** [comment]
Ok, I found where runtime is (you had 2 files: ZindexComponent.java and ZIndexComponent.java) and replaced 2nd with 1st. It successfully compiled, but still crashed the same way it crashed at first:
```
Download https://repo1.maven.org/maven2/com/badlogicgames/gdx/gdx-jnigen/1.0.0/gdx-jnigen-1.0.0.jar
Download https://repo1.maven.org/maven2/com/google/code/javaparser/javaparser/1.0.8/javaparser-1.0.8.jar
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
:processResources UP-TO-DATE
:classes
:run
Gtk-Message: Failed to load module "pantheon-filechooser-module"

(java:15247): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(java:15247): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(java:15247): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(java:15247): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(java:15247): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
Gtk-Message: Failed to load module "canberra-gtk-module"
java.lang.IllegalStateException: The Drawable has no context available.
	at org.lwjgl.opengl.DrawableGL.checkDestroyed(DrawableGL.java:152)
	at org.lwjgl.opengl.DrawableGL.makeCurrent(DrawableGL.java:109)
	at org.lwjgl.opengl.Display.makeCurrent(Display.java:706)
	at org.lwjgl.opengl.Display.makeCurrentAndSetSwapInterval(Display.java:1025)
	at org.lwjgl.opengl.Display.setParent(Display.java:452)
	at com.badlogic.gdx.backends.lwjgl.LwjglGraphics.setupDisplay(LwjglGraphics.java:137)
	at com.badlogic.gdx.backends.lwjgl.LwjglCanvas.create(LwjglCanvas.java:188)
	at com.badlogic.gdx.backends.lwjgl.LwjglCanvas$1.addNotify(LwjglCanvas.java:88)
	at java.awt.Container.addImpl(Container.java:1119)
	at java.awt.Container.add(Container.java:415)
	at com.badlogic.gdx.backends.lwjgl.LwjglFrame.addCanvas(LwjglFrame.java:135)
	at com.badlogic.gdx.backends.lwjgl.LwjglFrame$2.run(LwjglFrame.java:96)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:749)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:702)
	at java.awt.EventQueue$3.run(EventQueue.java:696)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:719)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
org.lwjgl.LWJGLException: X Error - disp: 0x7f79703f2eb0 serial: 247 error: BadWindow (invalid Window parameter) request_code: 2 minor_code: 0
	at org.lwjgl.opengl.LinuxDisplay.globalErrorHandler(LinuxDisplay.java:320)
	at org.lwjgl.opengl.LinuxKeyboard.nSetDetectableKeyRepeat(Native Method)
	at org.lwjgl.opengl.LinuxKeyboard.setDetectableKeyRepeat(LinuxKeyboard.java:152)
	at org.lwjgl.opengl.LinuxKeyboard.destroy(LinuxKeyboard.java:163)
	at org.lwjgl.opengl.LinuxDisplay.destroyKeyboard(LinuxDisplay.java:1264)
	at org.lwjgl.input.Keyboard.destroy(Keyboard.java:357)
	at org.lwjgl.opengl.Display.destroyWindow(Display.java:349)
	at org.lwjgl.opengl.Display.access$400(Display.java:65)
	at org.lwjgl.opengl.Display$5.destroy(Display.java:835)
	at org.lwjgl.opengl.Display.destroy(Display.java:1095)
	at com.badlogic.gdx.backends.lwjgl.LwjglGraphics.createDisplayPixelFormat(LwjglGraphics.java:197)
	at com.badlogic.gdx.backends.lwjgl.LwjglGraphics.setupDisplay(LwjglGraphics.java:174)
	at com.badlogic.gdx.backends.lwjgl.LwjglCanvas.create(LwjglCanvas.java:188)
	at com.badlogic.gdx.backends.lwjgl.LwjglCanvas$1.addNotify(LwjglCanvas.java:88)
	at java.awt.Container.addNotify(Container.java:2773)
	at javax.swing.JComponent.addNotify(JComponent.java:4749)
	at java.awt.Container.addNotify(Container.java:2773)
	at javax.swing.JComponent.addNotify(JComponent.java:4749)
	at java.awt.Container.addNotify(Container.java:2773)
	at javax.swing.JComponent.addNotify(JComponent.java:4749)
	at javax.swing.JRootPane.addNotify(JRootPane.java:757)
	at java.awt.Container.addNotify(Container.java:2773)
	at java.awt.Window.addNotify(Window.java:770)
	at java.awt.Frame.addNotify(Frame.java:487)
	at java.awt.Window.show(Window.java:1031)
	at java.awt.Component.show(Component.java:1654)
	at java.awt.Component.setVisible(Component.java:1606)
	at java.awt.Window.setVisible(Window.java:1014)
	at com.badlogic.gdx.backends.lwjgl.LwjglFrame$2.run(LwjglFrame.java:97)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:749)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:702)
	at java.awt.EventQueue$3.run(EventQueue.java:696)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:719)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at com.uwsoft.editor.Overlap2D.sendNotification(Overlap2D.java:107)
	at com.uwsoft.editor.Overlap2D.pause(Overlap2D.java:75)
	at com.badlogic.gdx.backends.lwjgl.LwjglCanvas$4.run(LwjglCanvas.java:308)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:749)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:702)
	at java.awt.EventQueue$3.run(EventQueue.java:696)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:719)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

BUILD SUCCESSFUL

Total time: 18.643 secs

```

**Comment by azakhary** [comment]
Okay so I fixed the z-index thing (if you just stash things and pull the latest it should be ok)

**Comment by azakhary** [comment]
Now back to this issue... hmmm

**Comment by azakhary** [comment]
I am curious at what point in application life cycle this happens. I think might be related to create if LWJGL app after splash. But had to debug on linux to figure it out.

**Comment by azakhary** [comment]
Also when searching for this in google:
Gtk-Message: Failed to load module "pantheon-filechooser-module"

You can see so many apps have this type of issue. Maybe you will be able to fix that on OS config level.

**Comment by vlaaad** [comment]
Looks like pantheon-filechooser-module is not related... At least so they say...

It may be an issue with my gpu drivers, because after I removed fglrx driver and brought back default open source driver... I started to receive another errors.

What I get is:

- `org.lwjgl.LWJGLException: X Error - disp: 0x7fe5a041b740 serial: 161 error: BadDrawable (invalid Pixmap or Window parameter) request_code: 154 minor_code: 7`
- `org.lwjgl.LWJGLException: X Error - disp: 0x7fe5a041b740 serial: 167 error: GLXBadWindow request_code: 155 minor_code: 32`
- `java.lang.IllegalArgumentException: Parameter 'directory' is not a directory`
- `Exception in thread "AWT-EventQueue-0" java.lang.RuntimeException: No OpenGL context found in the current thread.`

Googling this weird things usually leads to minecraft issue tracker where developers say that minecraft does not support that videocards.

Looks like it's more of a my notebook's problem than overlap2d's, so I will close this issue. 

Sadly, other libgdx apps (games I develop and, for example, Spine) work without problems.
